src/Microsoft.DotNet.XHarness.iOS.Shared/TestReporter.cs

```diff
@@ -0,0 +1,625 @@
+            }
+            else if (result.Succeeded)
+            {
+                mainLog.WriteLine(completionMessage);
+                Success = true;
+            }
+            else
+            {
+                mainLog.WriteLine(failureMessage);
+                Success = false;
+            }
+        }
+
+        public void LaunchCallback(Task<bool> launchResult)
+        {
+            if (launchResult.IsFaulted)
+            {
+                mainLog.WriteLine("Test launch failed: {0}", launchResult.Exception);
+            }
+            else if (launchResult.IsCanceled)
+            {
+                mainLog.WriteLine("Test launch was cancelled.");
+            }
+            else if (launchResult.Result)
+            {
+                mainLog.WriteLine("Test run started");
+            }
+            else
+            {
+                cancellationTokenSource.Cancel();
+                mainLog.WriteLine("Test launch timed out after {0} minute(s).", timeoutWatch.Elapsed.TotalMinutes);
+                timedout = true;
+            }
+        }
+
+        public async Task CollectSimulatorResult(Task<ProcessExecutionResult> processExecution)
+        {
+            isSimulatorTest = true;
+            await CollectResult(processExecution);
+
+            if (!Success.Value)
+            {
+                var (pid, launchFailure) = await GetPidFromRunLog();
+                this.launchFailure = launchFailure;
+                if (pid > 0)
+                {
+                    await KillAppProcess(pid, cancellationTokenSource);
+                }
+                else
+                {
+                    mainLog.WriteLine("Could not find pid in mtouch output.");
+                }
+            }
+        }
+
+        public async Task CollectDeviceResult(Task<ProcessExecutionResult> processExecution)
+        {
+            isSimulatorTest = false;
+            await CollectResult(processExecution);
+        }
+
+        async Task<(string ResultLine, bool Failed)> GetResultLine(string logPath)
+        {
+            string resultLine = null;
+            bool failed = false;
+            using var reader = new StreamReader(logPath);
+            string line = null;
+            while ((line = await reader.ReadLineAsync()) != null)
+            {
+                if (line.Contains("Tests run:"))
+                {
+                    Console.WriteLine(line);
+                    resultLine = line;
+                    break;
+                }
+                else if (line.Contains("[FAIL]"))
+                {
+                    Console.WriteLine(line);
+                    failed = true;
+                }
+            }
+            return (ResultLine: resultLine, Failed: failed);
+        }
+
+        async Task<(string resultLine, bool failed, bool crashed)> ParseResultFile(AppBundleInformation appInfo, string test_log_path, bool timed_out)
```

Comment:
can you provide a way to turn off the xml parsing magic? I don't think we want to use it in Helix.

Reply:
Is not used unless specified by the apprunner, but yes, we will provide a way not to parse it. The apprunner that @premun will do will not call it.

Reply:
So we just copy the outputs back and we're done?

Reply:
@premun yes, in the apprunner, just do not add the generation of the html, afaik is done there in xamarin-macios, then vsts/azure will pick the resulting xml. Makes the runner easier :)